pkg/ruler/ruler_test.go

```diff
@@ -213,6 +240,8 @@ func prepareRulerManager(t *testing.T, cfg Config, opts ...prepareOption) *Defau
 var _ MultiTenantManager = &DefaultMultiTenantManager{}
 
 func TestNotifierSendsUserIDHeader(t *testing.T) {
+	util_test.VerifyNoLeak(t)
```

Comment:
Shall we use `TestMain` and `VerifyNoLeakTestMain` to add this to all tests? (See https://pkg.go.dev/testing#hdr-Main)

Reply:
Good idea, done: https://github.com/grafana/mimir/pull/4975/commits/d450cc88aa7e2fdb968ed83696fe167cb8fbc8e2